### DISCUSSION TITLE
Why no Windows Binary Release?

### DISCUSSION THREAD

**Discussion by Suleman-Elahi** [owncast/owncast#3060] [Q&A]
As someone who doesn't have Linux or macOS, how will they use Owncast on Windows? Windows neither has Git, Go, GCC, FFmpeg. As maintainers and contributors, you should provide Windows builds.

Manual build fails as indicated in this issue already: https://github.com/owncast/owncast/issues/2814
but can be solved by building like this `set CGO_ENABLED=1 && go build`, given `gcc` is in PATH.

Another issue is with FFmpeg. It is looking for "FFmpeg" binary while the Windows doesn't know what the heck is that. FFmpeg in Windows is actually "FFmpeg.exe".

**Comment by rmens** [comment]
Sounds like more of a headache than it’s worth. Running server software on Windows is not that common as far as I know.

**Reply by Suleman-Elahi** [reply]
I agree but I use Windows most of the time so would be nice to have Owncast and since it is written Go, I can just run it directly just like Listmonk. But looks like I'll have to use WSL or Docker now. Also, Windows has Windows Server too if someone wants to host it on Windows then they might get into trouble too.

**Reply by gabek** [reply]
I'm pretty disconnected from the Windows world, but it sounds like everyone is running software under WSL and they seem pretty happy with that. But yeah, more headache than it's worth when there are already really good solutions people are already using.

That being said, if somebody was really passionate about Windows support and wanted to add it, I wouldn't be opposed.

**Reply by Suleman-Elahi** [reply]
I have managed to build it after an hour searching for how to enable CGO in Windows. Now, I it's Just stuck in "Unable to find FFmpeg" error. I think it needs FFMpeg binary as ffmpeg and not ffmpeg.exe.

![image](https://github.com/owncast/owncast/assets/24384996/1ab97fe8-afeb-4570-a3c4-b02b811acfac)

**Reply by Suleman-Elahi** [reply]
Got the reason for above error:

https://github.com/owncast/owncast/blob/develop/utils/utils.go#L241

Windows doesn't have `which` command. Instead, the `which` equivalent of Windows is `where`.

The FFmpeg issue is solved now after rebuilding, the different error arises. It is most likely caused by the invalid slashes in the paths, or I could be wrong.

![Screenshot (2)](https://github.com/owncast/owncast/assets/24384996/3dba77a8-2af5-439c-912b-db645177e3f0)
